### PR TITLE
Protobuf Format Support for Kafka Connector

### PR DESCRIPTION
**Pull request by saiharshavellanki**
- Added support for Protobuf format for Kafka connector
- Updated the documentation 

Screenshot 2024-06-29 at 8 13 00 AM
Screenshot 2024-06-29 at 8 13 09 AM

**Kafka topic parameters**
Screenshot 2024-06-29 at 8 14 21 AM

### CODE REVIEW

File: docs/pages/api-reference/connectors/kafka.md [hunk 1]
```diff
@@ -44,9 +44,9 @@ SASL password.
 The name of the kafka topic that needs to be sourced into the dataset.
 </Expandable>
 
-<Expandable title="format" type='"json" | Avro' defaultVal="json">
-The format of the data in Kafka topic. Both `"json"` and 
-[Avro](/api-reference/connectors/avro) supported.
+<Expandable title="format" type='"json" | Avro | Protobuf' defaultVal="json">
+The format of the data in Kafka topic. `"json"`, 
+[Avro](/api-reference/connectors/avro) and [Protobuf](/api-reference/connectors/protobuf) supported.
```
Comment (nikhilgarg28) [inline docs/pages/api-reference/connectors/kafka.md:49]: Does protobuf already work with Kinesis too or does it need a bit more work? If it already works, should we update similar documentation for Kinesis too?

File: docs/pages/api-reference/connectors/protobuf.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+</Expandable>
+
+<Expandable title="token" type="Optional[str]">
+Token to be used for authentication with the schema registry. Only one of 
+username/password or token must be provided.
+</Expandable>
+
+<pre snippet="api-reference/sources/kafka#kafka_with_protobuf"
+    status="success" message="Using protobuf registry with kafka"
+    highlight="13-18, 20">
```
Comment (nikhilgarg28) [inline docs/pages/api-reference/connectors/protobuf.md:45]: Request to not use this but instead use `docsnip-highlight <something>` tags inline with the code. That way, as code gets refactored, the highlight stays correct even if actual line numbers change.

File: fennel/connectors/connectors.py [hunk 3]
```diff
@@ -347,6 +347,14 @@ class Avro(BaseModel):
     token: Optional[str]
 
 
+class Protobuf(BaseModel):
+    registry: str
+    url: str
+    username: Optional[str]
+    password: Optional[str]
+    token: Optional[str]
```
Comment (nikhilgarg28) [inline fennel/connectors/connectors.py:355]: Just to confirm, at what level do we do error handling in scenarios:
1. One of username or password is given but not both
2. Token is given along with one or both of username/password. 

It'd be good ensure that doing something like that creates an error with clear message in the mock world itself (ideally within __init__ function itself, not even commit).

Comment (nikhilgarg28) [reply fennel/connectors/connectors.py:355]: Assuming this is done (already or after some revisions), could you please add tests to catch these?

**Review by nikhilgarg28 — CHANGES_REQUESTED** [review]
Requesting some changes

File: docs/pages/api-reference/connectors/kafka.md (hunk 1 above)
Comment (saiharshavellanki) [reply docs/pages/api-reference/connectors/kafka.md:49]: It doesn't work as of now. It needs a bit more work and tests. Post that we can add it to documentation

File: fennel/connectors/connectors.py (hunk 3 above)
Comment (saiharshavellanki) [reply fennel/connectors/connectors.py:355]: Added validation and some tests

**Review by nikhilgarg28 — APPROVED** [review]

**Pull request merged**